app/controllers/api/users.php

```diff
@@ -1150,21 +1160,23 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
 
         $email = \strtolower($email);
 
-        // Makes sure this email is not already used in another identity
-        $identityWithMatchingEmail = $dbForProject->findOne('identities', [
-            Query::equal('providerEmail', [$email]),
-            Query::notEqual('userId', $user->getId()),
-        ]);
-        if ($identityWithMatchingEmail !== false && !$identityWithMatchingEmail->isEmpty()) {
-            throw new Exception(Exception::USER_EMAIL_ALREADY_EXISTS);
-        }
+        if (\strlen($email) !== 0) {
+            // Makes sure this email is not already used in another identity
+            $identityWithMatchingEmail = $dbForProject->findOne('identities', [
+                Query::equal('providerEmail', [$email]),
+                Query::notEqual('userId', $user->getId()),
+            ]);
+            if ($identityWithMatchingEmail !== false && !$identityWithMatchingEmail->isEmpty()) {
+                throw new Exception(Exception::USER_EMAIL_ALREADY_EXISTS);
+            }
```

Comment:
Just like passwords, this is not safe and should not be allowed. The only way we can identify users is through their email. If people even use this endpoint by mistake, it means a specific user can never login if email is their only option..

Reply:
We can identify users without email. It's used for anonymous and phone sessions.